src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
```

Comment:
Oh wait. Do you mean to change the `try…finally` you already have to wrap the call to `SortedMap`? Yes, that will work. (Also, what does `Return()` do if passed `null`? Maybe just skip that test).